File: src/main/kotlin/io/mp3editor/util/StringUtil.java
package io.mp3editor.util;

/**
 * Contains util methods for String manipulations
 */
public class StringUtil {

    private static final String ILLEGAL_FILENAME_CHARACTERS = "[/:\\*\\?\"<>\\|\\\\]";

    /**
     * Clean str removing from it {@link StringUtil#ILLEGAL_FILENAME_CHARACTERS}
     *
     * @param str that need to be cleaned
     * @return cleaned {@code str}
     */
    public static String cleanString(String str) {
        return str.replaceAll(ILLEGAL_FILENAME_CHARACTERS, "");
    }
}

File: src/main/kotlin/io/mp3editor/util/StringUtil.kt
package io.mp3editor.util

/**
 * Contains util methods for String manipulations
 */
object StringUtil {
    private const val ILLEGAL_FILENAME_CHARACTERS = "[/:\\*\\?\"<>\\|\\\\]"

    /**
     * Clean str removing from it [StringUtil.ILLEGAL_FILENAME_CHARACTERS]
     *
     * @param str that need to be cleaned
     * @return cleaned `str`
     */
    @JvmStatic
    fun cleanString(str: String): String {
        return str.replace(ILLEGAL_FILENAME_CHARACTERS.toRegex(), "")
    }
}
